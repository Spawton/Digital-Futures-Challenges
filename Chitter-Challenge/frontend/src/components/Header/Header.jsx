import { NavLink } from "react-router-dom";

const Header = () => {
    return (

        <header>
            <nav className="navbar navbar-expand-lg ms-auto">
                <div className="container-fluid justify-content-start">
                    <h1 className="headerh4">CHITTER</h1>
                </div>
                <div className="container-fluid justify-content-end">
                    <NavLink to="/" className="headernavlink">Home</NavLink>
                    <NavLink to="/login" className="headernavlink">Log In</NavLink>
                    <NavLink to="/signup" className="headernavlink">Sign Up</NavLink>
                </div>
            </nav>
        </header >
    );
}

export default Header;