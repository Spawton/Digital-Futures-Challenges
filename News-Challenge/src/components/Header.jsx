import { NavLink } from "react-router-dom";

const Header = () => {
    return (
        <header>
            <nav>
                <h1>DFX News</h1>
                <NavLink to="/" className={isActive =>
                    "nav-link" + (!isActive ? " unselected" : "")}>
                    Home
                </NavLink>
            </nav>
        </header>
    );
}

export default Header;