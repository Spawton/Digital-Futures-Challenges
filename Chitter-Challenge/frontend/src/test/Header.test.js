import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import Header from '../components/Header/Header.jsx';

describe('Header tests', () => {
    it(`Header matches snapshot`, () => {
        const headerComponent = render(<Router><Header /></Router>);

        expect(headerComponent).toMatchSnapshot();
    });
})