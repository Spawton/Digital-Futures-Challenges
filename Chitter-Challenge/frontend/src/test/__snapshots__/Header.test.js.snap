// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Header matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="navbar navbar-expand-lg ms-auto"
        >
          <div
            class="container-fluid justify-content-start"
          >
            <h1
              class="headerh4"
            >
              CHITTER
            </h1>
          </div>
          <div
            class="container-fluid justify-content-end"
          >
            <a
              aria-current="page"
              class="headernavlink active"
              href="/"
            >
              Home
            </a>
            <a
              class="headernavlink"
              href="/login"
            >
              Log In
            </a>
            <a
              class="headernavlink"
              href="/signup"
            >
              Sign Up
            </a>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="navbar navbar-expand-lg ms-auto"
      >
        <div
          class="container-fluid justify-content-start"
        >
          <h1
            class="headerh4"
          >
            CHITTER
          </h1>
        </div>
        <div
          class="container-fluid justify-content-end"
        >
          <a
            aria-current="page"
            class="headernavlink active"
            href="/"
          >
            Home
          </a>
          <a
            class="headernavlink"
            href="/login"
          >
            Log In
          </a>
          <a
            class="headernavlink"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
